inc/class-wp-forms-api.php

```diff
@@ -876,9 +876,13 @@ static function process_element( $element, &$values, &$input ) {
 			$element['#value'] = isset( $input[$element['#key']] ) && $input[$element['#key']];
 		}
 		// Process checkbox by presence of #key, using the #checked value
-		// If not key is not set, set value to false
+		// If not key is not set or is empty, set value to false
 		else if ( $element['#type'] === 'checkbox' ) {
-			$element['#value'] = isset( $input[$element['#key']] ) ? $element['#checked'] : false;
+			if ( isset( $input[$element['#key']] ) && !empty( $input[$element['#key']] ) ) {
```

Comment:
All you really need is `!empty( $input[$element['#key']] )` since `empty()` acts as `isset()` as well